### PR TITLE
feat: first inquiry custom message

### PR DESCRIPTION
**Pull request by jo-rs**
The type of this PR is: Feature

This PR resolves [PURCHASE-2209]

### Description

This PR enables the user to submit a custom message through the first inquiry flow.
input_small

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[PURCHASE-2209]: https://artsyproduct.atlassian.net/browse/PURCHASE-2209

### CODE REVIEW

File: src/lib/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx [hunk 1]
```diff
@@ -213,6 +222,42 @@ const InquiryField = styled(Flex)`
...
+        onFocus={() => {
+          setBorderColor(color("purple100"))
+        }}
+        onBlur={() => {
+          setBorderColor(color("black10"))
+        }}
+        style={{ borderColor }}
+        numberOfLines={3}
+        multiline={true}
+      />
```
Comment (xtina-starr) [inline src/lib/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx:256]: Do we know how the `InputText` handles long messages? Like does the text area grow or scroll when it exceeds the space available? Also how does the 'numberOfLines` prop work?

Comment (jo-rs) [reply src/lib/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx:256]: I'll kill `numberOfLines` it only works on Android, I forgot to take that out.
Currently the input stays the same size, it scrolls on longer text.

File: src/lib/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx [hunk 2]
```diff
@@ -158,7 +159,9 @@ export const InquiryModal: React.FC<InquiryModalProps> = ({ artwork, ...props })
           resetAndExit()
         }}
         rightButtonText="Send"
-        rightButtonDisabled={state.inquiryQuestions.length === 0}
+        rightButtonDisabled={
+          (state.inquiryQuestions.length === 0 && !!questions.length) || (!questions.length && !state.message)
```
Comment (zephraph) [inline src/lib/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx:163]: I'm trying to work through this in my head, but I think this logic isn't correct. 

I think it's supposed to be as follows:

If no question is checked and the message is empty, disable the button.

Comment (zephraph) [reply src/lib/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx:163]: There are three factors considered here

1. Are there questions
2. Are any questions selected
3. Is there a message

In the situation that there are no questions we expect that there must be a message. If there's no message we expect that there must be questions selected. I think we can leverage those two facts to simplify the overall logic to just two factors.

```suggestion
          (state.inquiryQuestions.length === 0 && !state.message)
```

Comment (jo-rs) [reply src/lib/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx:163]: I think my brain might still not be in full working order; this took me a minute.

Yeah, you're right. Nice, @zephraph! 💪

File: src/lib/utils/ArtworkInquiry/ArtworkInquiryStore.tsx [hunk 3]
```diff
@@ -48,6 +49,11 @@ export const reducer = (
         ...inquiryState,
         inquiryQuestions: newSelection,
       }
+    case "setMessage":
+      return {
+        ...inquiryState,
+        message: action.payload,
```
Comment (zephraph) [inline src/lib/utils/ArtworkInquiry/ArtworkInquiryStore.tsx:55]: I think it'd be good for us to remove any extra whitespace from the payload to the user isn't just submitting with a lot of whitespace or that there's weird padding in the email that goes out to partners. 

```suggestion
        message: action.payload.trim(),
```

File: src/lib/utils/ArtworkInquiry/ArtworkInquiryStore.tsx [hunk 4]
```diff
@@ -52,7 +52,7 @@ export const reducer = (
     case "setMessage":
       return {
         ...inquiryState,
-        message: action.payload,
+        message: action.payload.trim(),
```
Comment (xtina-starr) [inline src/lib/utils/ArtworkInquiry/ArtworkInquiryStore.tsx:55]: 👌

**Review by zephraph — APPROVED** [review]
This looks good to me. Thanks for seeing this through!

**Comment by xtina-starr** [comment]
I agree! I'm going to add the `SquashOnGreen` label.

File: src/lib/utils/ArtworkInquiry/ArtworkInquiryStore.tsx (hunk 4 above)
Comment (jo-rs) [reply src/lib/utils/ArtworkInquiry/ArtworkInquiryStore.tsx:55]: Actually, @zephraph, this cuts spaces on every keyup. So, I'll modify to trim only on submit.

Comment (jo-rs) [reply src/lib/utils/ArtworkInquiry/ArtworkInquiryStore.tsx:55]: @xtina-starr the `trim` actually doesn't work the way it is now. I'll submit another PR with a fix.

**Comment by jo-rs** [comment]
Bugfix on [this](https://github.com/artsy/eigen/pull/4165) PR

**Pull request merged**